src/Codeception/Lib/Generator/Actor.php

```diff
@@ -236,5 +235,45 @@ public function getNumMethods()
         return $this->numMethods;
     }
 
+    private function getDefaultValue(\ReflectionParameter $param)
+    {
+        $isPlainArray = function ($value) {
+            return ((count($value) === 0)
+                || (
+                    (array_keys($value) === range(0, count($value) - 1))
+                    && (0 === count(array_filter(array_keys($value), 'is_string'))))
+            );
+        };
+        if ($param->isDefaultValueAvailable()) {
+            if (method_exists($param, 'isDefaultValueConstant ') && $param->isDefaultValueConstant()) {
+                $constName = $param->getDefaultValueConstantName();
+                if (false !== strpos($constName, '::')) {
+                    list($class, $const) = explode('::', $constName);
+                    if (in_array($class, ['self', 'static'])) {
+                        $constName = $param->getDeclaringClass()->getName().'::'.$const;
+                    }
+                }
+
+                return $constName;
+            }
 
+            $paramValue = $param->getDefaultValue();
+            switch (true) {
```

Comment:
I think you are right!

Only possible difference is the array and string formatting.
An array gets put into one line and gets simplified if it was a plain simple one dimensional array.
In the case of string, it is escaped; all escape sequences get converted to ASCII by `var_export` as well as `print_r`.

I can change it if you prefer.

I think that function could do with some unit testing. Is there a specific way doing unit tests in codeception?